flink-connectors/flink-connector-kinesis/src/test/java/org/apache/flink/streaming/connectors/kinesis/util/AwsV2UtilTest.java

```diff
@@ -0,0 +1,407 @@
+	}
+
+	@Test
+	public void testGetWebIdentityTokenFileCredentialsProviderWithWebIdentityFile() {
+		Properties properties = properties(AWS_CREDENTIALS_PROVIDER, "WEB_IDENTITY_TOKEN");
+		properties.setProperty(webIdentityTokenFile(AWS_CREDENTIALS_PROVIDER), "webIdentityTokenFile");
+
+		WebIdentityTokenFileCredentialsProvider.Builder builder = mockWebIdentityTokenFileCredentialsProviderBuilder();
+
+		AwsV2Util.getWebIdentityTokenFileCredentialsProvider(builder, properties, AWS_CREDENTIALS_PROVIDER);
+
+		verify(builder).webIdentityTokenFile(Paths.get("webIdentityTokenFile"));
+	}
+
+	@Test
+	public void testGetCredentialsProviderAuto() {
+		Properties properties = properties(AWS_CREDENTIALS_PROVIDER, "AUTO");
+
+		AwsCredentialsProvider credentialsProvider = AwsV2Util.getCredentialsProvider(properties);
+
+		assertTrue(credentialsProvider instanceof DefaultCredentialsProvider);
+	}
+
+	@Test
+	public void testGetCredentialsProviderAssumeRole() {
+		Properties properties = spy(properties(AWS_CREDENTIALS_PROVIDER, "ASSUME_ROLE"));
+		properties.setProperty(AWS_REGION, "eu-west-2");
+
+		AwsCredentialsProvider credentialsProvider = AwsV2Util.getCredentialsProvider(properties);
+
+		assertTrue(credentialsProvider instanceof StsAssumeRoleCredentialsProvider);
+
+		verify(properties).getProperty(AWSConfigConstants.roleArn(AWS_CREDENTIALS_PROVIDER));
+		verify(properties).getProperty(AWSConfigConstants.roleSessionName(AWS_CREDENTIALS_PROVIDER));
+		verify(properties).getProperty(AWSConfigConstants.externalId(AWS_CREDENTIALS_PROVIDER));
+		verify(properties).getProperty(AWS_REGION);
+	}
+
+	@Test
+	public void testGetCredentialsProviderBasic() {
+		Properties properties = properties(AWS_CREDENTIALS_PROVIDER, "BASIC");
+		properties.setProperty(AWSConfigConstants.accessKeyId(AWS_CREDENTIALS_PROVIDER), "ak");
+		properties.setProperty(AWSConfigConstants.secretKey(AWS_CREDENTIALS_PROVIDER), "sk");
+
+		AwsCredentials credentials = AwsV2Util.getCredentialsProvider(properties).resolveCredentials();
+
+		assertEquals("ak", credentials.accessKeyId());
+		assertEquals("sk", credentials.secretAccessKey());
+	}
+
+	@Test
+	public void testGetCredentialsProviderProfile() {
+		Properties properties = properties(AWS_CREDENTIALS_PROVIDER, "PROFILE");
+		properties.put(AWSConfigConstants.profileName(AWS_CREDENTIALS_PROVIDER), "default");
+		properties.put(AWSConfigConstants.profilePath(AWS_CREDENTIALS_PROVIDER), "src/test/resources/profile");
+
+		AwsCredentialsProvider credentialsProvider = AwsV2Util.getCredentialsProvider(properties);
+
+		assertTrue(credentialsProvider instanceof ProfileCredentialsProvider);
+
+		AwsCredentials credentials = credentialsProvider.resolveCredentials();
+		assertEquals("11111111111111111111", credentials.accessKeyId());
+		assertEquals("wJalrXUtnFEMI/K7MDENG/bPxRfiCY1111111111", credentials.secretAccessKey());
+	}
+
+	@Test
+	public void testGetCredentialsProviderNamedProfile() {
```

Comment:
Should we just throw an exception when the `CredentialsProvicer` is set to an illegal name, rather than give it a default value?

Reply:
I am not sure what you mean, can you please elaborate? It already throws an exception if the `CredentialProvider` is not defined in the enum or supported by the `getCredentialsProvider` method.